src/common/main.lua

```diff
@@ -188,16 +190,71 @@ function ant.init()
 	end)
 
 	createActionHandler(ActionMap.SetRecord, function(msg)
-		utils.assertHasPermission(msg.From)
-
-		local name = string.lower(msg.Tags["Sub-Domain"])
-		local transactionId = msg.Tags["Transaction-Id"]
-		local ttlSeconds = tonumber(msg.Tags["TTL-Seconds"])
-		local priority = tonumber(msg.Tags["Priority"])
+		local name = string.lower(msg["Sub-Domain"])
+		local transactionId = msg["Transaction-Id"]
+		local ttlSeconds = tonumber(msg["TTL-Seconds"])
+		local priority = tonumber(msg["Priority"])
```

Comment:
question: what are the ramifications of undername owners being able to set priority? what if there are collisions?

suggestion: `priority` should be a protected field that can only be modified by controllers/owners

Reply:
If priority is nil we validate that they are having record permission, if its not nil we verify that they are the owner/controllers of the ant (see below, ln:205)

You are right to question the impact there. If undername owners can set the priority they can undercut other domains and affect how others are served. Its probably not a huge deal but we decided it was not worth having undername owners have the ability to set it.